src/WingetCreateCLI/GitHubOAuth.cs

```diff
@@ -65,7 +70,10 @@ public static string ReadTokenCache()
         /// <param name="token">Token to be cached.</param>
         public static void WriteTokenCache(string token)
         {
-            var bytes = ProtectedData.Protect(Encoding.UTF8.GetBytes(token), EntropyBytes, DataProtectionScope.CurrentUser);
+            var bytes = Encoding.UTF8.GetBytes(token);
+#if WINDOWS
+            bytes = ProtectedData.Protect(Encoding.UTF8.GetBytes(token), EntropyBytes, DataProtectionScope.CurrentUser);
```

Comment:
We can use libsecret and keychain on Linux and macOS respectively. Can we merge this PR but not release winget-create for Linux and macOS platforms until this has been fixed? I will implement this in another PR because  this PR has already become large enough.